src/core/immutableMap.ts

```diff
@@ -71,6 +72,21 @@ export const mKeys: MKeys = <K, V>(m: ImmutableMap<K, V>) => {
   return keys
 }
 
+export const mForEach: MForEach = <K, V>(
+  map: ImmutableMap<K, V>,
+  cb: (a: K) => void
+) => {
+  const cache = new Map()
```

Comment:
I think we can use `Set` instead of `Map`.
(It can be even be `WeakSet`, which can be more performant, but not sure.)

I'd name it `seen` or something. `const seen = Set<K>()`